rollup/fees/rollup_fee.go

```diff
@@ -203,11 +205,22 @@ func calculateEncodedRollupFeeFeeFeynman(
 	execScalar *big.Int,
 	blobScalar *big.Int,
 ) *big.Int {
-	// tx size (RLP-encoded)
 	txSize := big.NewInt(int64(len(data)))
 
-	// compression_ratio(tx) = 1 (placeholder, scaled to match scalars precision)
-	compressionRatio := big.NewInt(rcfg.Precision.Int64())
+	// Default compression ratio is 1.0 (no compression)
+	compressionRatioInt := big.NewInt(rcfg.Precision.Int64())
+
+	if len(data) != 0 {
+		compressedBytes, err := zstd.CompressScrollBatchBytes(data)
```

Comment:
I think we can just use any go zstd lib, don't need to use our `da-codec`, what do you think? @colinlyguo

Reply:
Waiting on this

Reply:
What are the motivations to not use our own zstd lib? the 128 bytes overhead?

Reply:
Please sync with Colin, this is something we want to do in other codebases too. Circuits can now support the whole zstd spec so we don't need to hardcode using a specific implementation (true with some caveats).

Reply:
Yeah. We can use any Go Zstd lib. And a reminder is that the spec can also be used in Reth, for constructing historical blocks from DA>

Reply:
A historically symbol conflict issue: https://scrollco.slack.com/archives/C02PKAQ2430/p1720694878378479

Reply:
Pure Go implementation, such as https://github.com/klauspost/compress/tree/master/zstd#zstd won't have any issues with C's global symbol conflicts when loading multiple zstd libraries. But I'm afraid Rust cannot have an identical implementation to reproduce the same compression ratio.

Also, pls double-check if a version of zstd running on different operating systems will produce the same results.

Reply:
zstd is [deterministic given same input, same set of parameters and same version.](https://github.com/facebook/zstd/issues/2949)